Cogs.Publishers/JsonPublisher.cs

```diff
@@ -0,0 +1,111 @@
+
+namespace Cogs.Publishers
+{
+    public class JsonPublisher
+    {
+        private JsonSerializerSettings settings= new JsonSerializerSettings();
+
+        public string CogsLocation { get; set; }
+        public string TargetDirectory { get; set; }
+        public bool Overwrite { get; set; }
+
+        public string TargetNamespace { get; set; } = "ddi:3_4";
+
+        //Dictionary<string, string> createdElements = new Dictionary<string, string>();
+
+        public void Publish(CogsModel model)
+        {
+            if (CogsLocation == null)
+            {
+                throw new InvalidOperationException("Cogs location must be specified");
+            }
+            if (TargetDirectory == null)
+            {
+                throw new InvalidOperationException("Target directory must be specified");
+            }
+
+            if (Overwrite && Directory.Exists(TargetDirectory))
+            {
+                Directory.Delete(TargetDirectory, true);
+            }
+
+            Directory.CreateDirectory(TargetDirectory);
+            settings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+            settings.Formatting = Formatting.Indented;
+            settings.Converters.Add(new JsonScehmaPropConverter());
+            settings.DefaultValueHandling = DefaultValueHandling.Ignore;
+
+            //create a list to store jsonschema for each itemtype
+            List<JsonSchema> items = new List<JsonSchema>();
+
+            //foreach (ItemType item in model.ItemTypes)
+            //{
+            //   Console.WriteLine(JsonConvert.SerializeObject(item, settings));
+            //}
+            foreach (ItemType item in model.ItemTypes)
+            {
+                JsonSchema temp = new JsonSchema();
+                temp.title = item.Name;                          //get the name of the itemtype
+                temp.type = "object";                           //get the type of the itemtype which is usually Object
+                temp.properties = new List<JsonSchemaProp>();
+                temp.required = new List<string>();
+                if (item.ExtendsTypeName != "")             //Check whether there it extends another class
+                {
+                    //get the Parent information
+                    if (item.ParentTypes != null)
+                    {
+                        //traverse parent list, find the properties of the parents
+                        foreach (var properti in item.ParentTypes)
+                        {
+                            if (properti.Properties != null)
+                            {
+                                //traverse the properties and get all the information regarding variable. 
+                                foreach (var inner_prop in properti.Properties)
+                                {
+                                    var parentprop = new JsonSchemaProp();
+                                    parentprop.name = inner_prop.Name;
```

Comment:
Lets refactor getting the properties from the parents and the items, so the property creation isn't repeated.

We could make a local list of Properties, add the Properties from the parents and the item in the correct order, and then convert them all at once to JsonSchemaProps